02-starting-with-data.Rmd

```diff
@@ -111,10 +111,132 @@ Based on the output of `str(surveys)`, can you answer the following questions?
 ```
 --->
 
-As you can see, many of the columns consist of integers, however, the columns
-`species` and `sex` are of a special class called a `factor`. Before we learn
-more about the `data.frame` class, let's talk about factors. They are very
-useful but not necessarily intuitive, and therefore require some attention.
+## Inspecting `data.frame` Objects
+
+We already saw how the functions `head()` and `str()` can be useful to check the
+content and the structure of a `data.frame`. Here is a non-exhaustive list of
+functions to get a sense of the content/structure of the data.
+
+* Size:
+    * `dim()` - returns a vector with the number of rows in the first element,
+          and the number of columns as the second element (the **dim**ensions of
+          the object)
+    * `nrow()` - returns the number of rows
+    * `ncol()` - returns the number of columns
+
+* Content:
+    * `head()` - shows the first 6 rows
+    * `tail()` - shows the last 6 rows
+
+* Names:
+    * `names()` - returns the column names (synonym of `colnames()` for `data.frame`
+	   objects)
+    * `rownames()` - returns the row names
+
+* Summary:
+    * `str()` - structure of the object and information about the class, length and
+	   content of  each column
+    * `summary()` - summary statistics for each column
+
+Note: most of these functions are "generic", they can be used on other types of
+objects besides `data.frame`.
+
```

Comment:
Might want to add the code here for actually doing these things, e.g. str(surveys), so that instructors know they should be typing these with learners rather than just talking about them.